### ISSUE TITLE
/setup Error 500, Blank Page

### ISSUE THREAD

**Issue by TheMikeAmbrose** [CachetHQ/Cachet#1684]
Hey there,

I'm having an issue getting the /setup page to appear properly after installing Cachet. I'm using CentOS 6.7, Apache, PHP 7.0 and all of the required dependencies are installed. I've installed the application properly and it wrote the initial SQL tables to the database fine. When going in browser to the domain it's behind, I get a generic Error 500 page below:
https://i.gyazo.com/f5050abd8bf1220819ee680e9f3f00ce.png

I did chmod recursive the storage directory to 777, this didn't fix the issue.

Next I removed all files from the boostrap/cache directory, after refreshing the page, it no longer goes to the Error 500 page but to a blank white page with nothing on it. I checked the Apache error_log and this error is in there (This ONLY is put in the error_log after deleting bootstrap/cache/*, if the files are in the cache directory, the generic Error 500 page appears):

`[Mon Apr 04 06:06:16 2016] [error] [client (redacted)] PHP Fatal error:  Uncaught Illuminate\\Contracts\\Container\\BindingResolutionException: Unresolvable dependency resolving [Parameter #0 [ <required> $path ]] in class GrahamCampbell\\Exceptions\\ExceptionInfo in /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php:839\nStack trace:\n#0 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(806): Illuminate\\Container\\Container->resolveNonClass(Object(ReflectionParameter))\n#1 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(777): Illuminate\\Container\\Container->getDependencies(Array, Array)\n#2 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(633): Illuminate\\Container\\Container->build('GrahamCampbell\\\\...', Array)\n#3 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(697): Illuminate\\Container\\Container->make('GrahamCampbell\\\\...', Array)\n#4 /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php(853): Illuminate\\Foun in /var/www/Cachet/vendor/laravel/framework/src/Illuminate/Container/Container.php on line 839, referer: http://(redactedA)/setup`

Any suggestions would be appreciated.

**Comment by jbrooksuk** [comment]
Did you run `php artisan app:install`?

**Comment by jbrooksuk** [comment]
Can we see your `.env` file please (remember to hide any sensitive information.)

**Comment by TheMikeAmbrose** [comment]
Yes, I ran php atrisan app:install a few times, cleared the DB and tried again.

Here is the .env:

`APP_ENV=production
APP_DEBUG=false
APP_URL=http://status.(removed).com
APP_KEY=base64:(removed)

DB_DRIVER=mysql
DB_HOST=localhost
DB_DATABASE=cachet
DB_USERNAME=cachet
DB_PASSWORD=(removed)
DB_PORT=3306
DB_PREFIX=null

CACHE_DRIVER=file
SESSION_DRIVER=file
QUEUE_DRIVER=database
CACHET_EMOJI=false

MAIL_DRIVER=mail
MAIL_HOST=null
MAIL_PORT=null
MAIL_USERNAME=null
MAIL_PASSWORD=null
MAIL_ADDRESS=null
MAIL_NAME=null
MAIL_ENCRYPTION=tls

REDIS_HOST=null
REDIS_DATABASE=null
REDIS_PORT=null

GITHUB_TOKEN=null
`

**Comment by jbrooksuk** [comment]
`APP_KEY` shouldn't be base64, should it?

**Comment by TheMikeAmbrose** [comment]
Everything after APP_KEY was generated and put in it automatically after running "php artisan key:generate"

[root@x Cachet]# php artisan key:generate
Application key [base64:YtEst6(removed...)Vk=] set successfully.

**Comment by jbrooksuk** [comment]
That's weird. I assume it's due to lacking a dependency, ping @GrahamCampbell?

**Comment by GrahamCampbell** [comment]
The base64 thing is something new that Taylor added quite recently James. It's so the keys can use the whole character space instead of just alphanumeric.

**Comment by jbrooksuk** [comment]
@GrahamCampbell ah cool, I didn't know about that. Nice!

Do you know what might be causing @TheMikeAmbrose's issue?

**Comment by GrahamCampbell** [comment]
Looks like the config folder is out of sync.

**Comment by GrahamCampbell** [comment]
Maybe try restarting php-fpm.

**Comment by TheMikeAmbrose** [comment]
Restarting php-fpm didn't help.

**Comment by jbrooksuk** [comment]
What about running:

- `php artisan config:cache`.
- `rm -rf bootstrap/cache/*`

In that order please :)

**Comment by TheMikeAmbrose** [comment]
Running those didn't change anything, same errors.

If I can't find what's causing it I'll just switch over to Ubuntu and use it on there, I've had hell of a time getting anything using larvel or nodejs to work properly on CentOS 6, while everything runs fine in Ubuntu out of the box (speaking generally, just about Cachet itself.)

**Comment by jbrooksuk** [comment]
I setup Cachet for somebody on CentOS once. Took me ages.

Ubuntu will be much faster if you're able to use it :)

**Comment by TheMikeAmbrose** [comment]
Nope, same exact errors on Ubuntu 15 64x, tried v2.2.0 and v2.1.0. Also used PHP 5.6 instead of PHP 7.0 this time.

**Comment by jbrooksuk** [comment]
You've followed all the commands and docs?

**Comment by TheMikeAmbrose** [comment]
Unless there's more to it than the Prerequisites and then https://docs.cachethq.io/docs/installing-cachet, yes.

**Comment by bjshively** [comment]
@TheMikeAmbrose I've had this issue before due to permissions. After smashing many different combinations into my keyboard, I eventually got it working. This might not be ideal from a security perspective, but to troubleshoot, I'd try (from the Cachet root dir) `chmod +777 ./storage` and if you have a ./database `chmod +777 ./database`.

**Comment by TheMikeAmbrose** [comment]
Here's the laravel log from when the page is loaded:
http://hastebin.com/usoresaqaj.pas

Tried it on Ubuntu 14.04 and CentOS 7 also.

**Comment by jbrooksuk** [comment]
Ping @GrahamCampbell again.

**Comment by GrahamCampbell** [comment]
@TheMikeAmbrose Can you comment out all the filters and all but the first displayer in the exceptions.php config (just for now), then try again. That'll allow us to see the actual error here.

**Comment by TheMikeAmbrose** [comment]
I commented all out except the first and nothing changed, tried a few others and just left the debug one uncommented and tried numerous others and it all has the same white page, same errors in larvel log also as last times.

Are you sure there's no dependencies missing from the doc page? I've literally tried this on half of the linux distros that are out there now with PHP 5.5, 5.6, 7.0. All of the prerequisites are installed and I've tried three versions via git from the doc page and the zip source from 9 days ago.
